src/main/java/de/tum/in/www1/artemis/service/ExerciseService.java

```diff
@@ -520,6 +520,9 @@ public void filterForCourseDashboard(Exercise exercise, List<StudentParticipatio
             if (submission != null) {
                 participation.setSubmissions(Set.of(submission));
             }
+            else {
```

Comment:
Could be shortened to `submission ? participation.setSubmissions(Set.of(submission)) : participation.setSubmissions(null);` I guess.

Reply:
submission != null ? participation.setSubmissions(Set.of(submission)) : participation.setSubmissions(null);

😉

Reply:
submission != null ? participation.setSubmissions(Set.of(submission)) : participation.setSubmissions(null);

😉

Reply:
no it can't because it is not a statement or variable assignment. 

![image](https://user-images.githubusercontent.com/33484318/107515724-af538c80-6bab-11eb-82dd-f35083a7e985.png)

Reply:
You could write `participation.setSubmissions(submission != null ? ... : ...)` but Imo it is ok as if else.

Reply:
this would work: 

```
participation.setSubmissions(submission != null ? Set.of(submission) : null);
```